### DISCUSSION TITLE
Multirow figure gets cutoff when generating a PDF

### DISCUSSION THREAD

**Discussion by russ-dvm** [quarto-dev/quarto-cli#5583] [Q&A] [answered]
First off, this is a cross-post from Stackoverflow where this question received no answers... Apologies if that is a faux-pas.

I am creating a PDF using Quarto. I have several instances in which I am using [subfigures](https://quarto.org/docs/authoring/figures.html#subfigures). The subfigures are large, and have to remain large - each component of the subfigure should be approximately the size of the page. The problem is that when rendering to PDF, the second subfigure does not float to the next page - instead it is placed below the first. This seems similar to issue #5553 but here I include a reproducible example, using images taken from this Wikipedia article downloaded at original resolution.

https://en.wikipedia.org/wiki/Petasites_hybridus

````qmd
---
title: "test"

format:
  html:
    theme: cosmo
  pdf:
    documentclass: scrreprt
    include-in-header: 
      text: |
        \usepackage{booktabs}
        \usepackage{longtable}
        \usepackage{array}
        \usepackage{multirow}
        \usepackage{wrapfig}
        \usepackage{float}
        \usepackage{colortbl}
        \usepackage{pdflscape}
        \usepackage{threeparttable}
        \usepackage{threeparttablex}
        \usepackage[normalem]{ulem}
        \usepackage{makecell}
        \usepackage{xcolor}
  docx: default
editor: visual
---

test

::: {#fig-flower layout-nrow="2"}
![](images/Petasites_hybridus_inflorescence_-_Keila.jpg)

![](images/Petasites_hybridus_leaf.JPG)
:::

````

Result:
![NUWVa](https://github.com/quarto-dev/quarto-cli/assets/17009528/0519b12b-a033-4302-b872-7bc7acf6632e)

Any thoughts on how to fix this? Thanks!

**Comment by mcanouil** [comment]
Since this is mostly LaTeX, you can add `keep-tex: true` to your frontmatter, take a look at the LaTeX code that produced this.

Note that "responsive" image is not a thing in LaTeX, so if you include images that are too large to fit in a page, that's basically up to you to resize the figures (using markup/LaTeX code or figure editor).

---

Could you share a small "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by russ-dvm** [reply]
Is the above not suitable? It is a working Quarto example.

**Reply by mcanouil** [reply]
For this to work, anyone needs to go find the images, download them, put them into a "images" directory, thus this is not self-contained/reproducible.

It's really not to bother you, but I (and other) prefer spending time on addressing the question rather than spending time on trying to reproduce or make an example.

**Reply by russ-dvm** [reply]
I absolutely understand and agree - wasn't sure how else to include images in the example.

Here is a repository with the code and images.

https://github.com/russ-dvm/qmd-example

**Reply by cscheid** [reply, marked as answer]
I don't think it's possible for us to make "multi-page subfigures" that float like you expect them to. That requires coordination between multiple different floats in LaTeX. From my understanding, that's not how the LaTeX layout system works.

**Reply by mcanouil** [reply]
Indeed, in LaTeX, it's not that easy to get floats behave as in HTML with CSS.
Most of the time, you end tweaking each floats (tables or figures) in custom ways.

That's why here the easiest path, is to resize the images to fit in the width of the page (I find it weird to have a "column"-wise subfigure, _i.e._, I never encountered such thing).
Also, in my opinion, if you know your images can't fit in one page, why try to do that?

**Reply by russ-dvm** [reply]
Thank you both for your replies. It sounds as if I am trying to fit a square peg in a round hole when it comes to LaTeX floats. 

In response to your specific question, I am not trying to fit two images on one page - that's the exact problem I am trying to overcome. Your suggestion of resizing the images is unfortunately not suitable, as the images contain text that needs to be legible, and resizing them so that >1 image fits on a page will render the text too small. The column-wise subfigure is unusual, I agree, but it was my attempt to find a creative solution to my problem - obviously reflecting a poor understanding of how LaTeX works (but, FWIW, providing a perfectly fine solution in HTML).

Thank you both for your insights. I'll try and figure out a different approach.

**Reply by mcanouil** [reply]
Use two figures/floats (no layout/fenced div), LaTeX should move the second figure to where there is room for it, thus on the next page.
Subfigures should be together, if they are not, it means they are figures not subfigures.

**Comment by CMFell** [comment]
I have essentially the same question (I think) but in a slightly different context. I am trying to create both revealjs slides and pdf from the same document. I have multiple subfigures as fragments so that in the revealjs slides they stack and replace each other on the slide, this works fine. When it comes to pdf though they are all printed on the same page one after each other and do not float so later subfigures get cut off. 

From what I understand from the answer above the only way is to put them as separate figures which means the revealjs stacking doesn't work. Is there a way to get the subfigures to split onto multiple pages or else achieve the stacking effect with separate figures?

````
---
title: fragment test
format:
  pdf: 
    fig-height: 5
    fig-dpi: 96
    fig-pos: "H"
  html: default
  revealjs:
    output-file: fragmentspdf-revealjs.html
---

::::: {#fig-carsplots}

:::: {.r-stack}

::: {.fragment}

```{r}
#| warning: false
#| echo: false
library(tidyverse)
cars1 <- ggplot(mtcars) + geom_point(aes(x=cyl, y=mpg))
ggsave("cars1.png", cars1)
cars1
```

:::

::: {.fragment}

```{r}
#| warning: false
#| echo: false
cars2 <- ggplot(mtcars) + geom_point(aes(x=cyl, y=mpg)) +
  geom_vline(aes(xintercept=5), color='red')
ggsave("cars2.png", cars2)
cars2

```

:::

::: {.fragment}

```{r}
#| warning: false
#| echo: false
cars3 <- ggplot(mtcars) + geom_point(aes(x=cyl, y=mpg)) +
  geom_vline(aes(xintercept=5), color='red') +
  geom_vline(aes(xintercept=7), color='blue') 
ggsave("cars3.png", cars3)
cars3

```

:::

::::

MPG vs cylinders

:::::

## Next slide

:::: {#fig-carsmpg2}

::: {.r-stack}

![](cars1.png){.fragment}


![](cars2.png){.fragment}


![](cars3.png){.fragment}

![](cars3.png){.fragment}


:::

MPG vs cylinders from pictures

::::
````

In my actual document I do this with both pictures and generated outputs and have the same problems with both methods, so I've included examples of both in the example.

**Reply by mcanouil** [reply]
You can investigate by looking at the LaTeX code produced using `keep-tex: true` in your YAML front matter.

Why would you expect the same layout to behave differently using images from computation or from path? Take a look at the intermediate markdown using `keep-md: true`.

**Reply by mcanouil** [reply]
Here is a simpler version (1.5 required):

````qmd
---
title: fragment test
format:
  pdf: default
  revealjs: default
---

## Next slide

:::: {#fig-carsmpg2}

::: {.r-stack}

[{{< placeholder 100 200 >}}]{.fragment}

[{{< placeholder 100 200 >}}]{.fragment}

[{{< placeholder 100 200 >}}]{.fragment}

[{{< placeholder 100 200 >}}]{.fragment}

:::

MPG vs cylinders from pictures

::::
````

Using `:::: {#fig-carsmpg2}` to enclose the stack of figure means you want the whole stack to be considered as one single object, in other you are asking for individual figures to not float.

So, here everything behaves as expected.

Basically the marked answer is still the answer: https://github.com/quarto-dev/quarto-cli/discussions/5583#discussioncomment-5931410

**Comment by amael-ls** [comment]
I actually have a similar question, but with computed figures. Here is a MWE:

`````qmd
---
title: "MWE"
keep-tex: true
format:
  pdf:
    mathspec: true
---
See @fig-test
```{r}
#| label: fig-test
#| fig-cap: Random plots
#| fig-subcap: 
#|   - Random 1
#|   - Random 2
#|   - Random 3
#|   - Random 4
#|   - Random 5

for (i in 1:5)
	plot(rnorm(100), pch = 19, cex = 0.5)
```
`````

I can of course use `keep-tex: true` and then add `\ContinuedFloat` in the tex file, but is it not very practical. Maybe it would be possible to create a lua filter, and then mix latex code with R code (which is not possible without a lua filter as stated [here](https://github.com/quarto-dev/quarto-cli/discussions/6736)).

For completeness, I add the link to the [stack overflow question](https://stackoverflow.com/questions/76167424/multirow-figure-gets-cutoff-when-generating-a-pdf-via-quarto/79492671#79492671) of @russ-dvm

I added a specific question on [stack overflow](https://stackoverflow.com/questions/79492719/computed-subfigure-too-large) for computed figures
